lib/matplotlib/lines.py

```diff
@@ -1213,6 +1242,25 @@ def set_markersize(self, sz):
             self.stale = True
         self._markersize = sz
 
+    def set_secondcolor(self, secondcolor):
+        """
+        Set a second color to fill the gaps in the dashed line style.
+
+        .. warning::
+
+            In some cases the two line colors may overlap, which could lead to
+            unwanted effects if also setting ``alpha``.
```

Comment:
I think a note is sufficient, and let's give some more context:

```suggestion
        .. note::

            Striped lines are created by drawing two interleaved dashed lines.
            There can be single-pixel sized overlaps between those two, which
            may result in artifacts when using transparency.
```

Reply:
I just removed "single-pixel sized" from your suggestion, as the [test image](https://github.com/matplotlib/matplotlib/pull/23208/files#diff-48e9b056c733b9b6310cebf72269a4d251c34fb49bcb731fe5eaa26349794339) shows the overlaps may be larger than that in some cases.